_posts/en/newsletters/2024-01-17-newsletter.md

```diff
@@ -0,0 +1,246 @@
+---
+title: 'Bitcoin Optech Newsletter #285'
+permalink: /en/newsletters/2024/01/17/
+name: 2024-01-17-newsletter
+slug: 2024-01-17-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter FIXME
+
+## News
+
+- **Disclosure of past vulnerability in Core Lightning:** Matt Morehouse
+  used Delving Bitcoin to [announce][morehouse delving] a vulnerability
+  he had previously [responsibly disclosed][topic responsible
+  disclosures] that affected Core Lightning versions 23.02 through
+  23.05.2.  More recent versions of 23.08 or higher are not affected.
+
+    The new vulnerability was discovered by Morehouse's following up on
+    his prior work on fake funding, which he also responsibly disclosed
+    (see [Newsletter #266][news266 lnbugs]).  When re-testing nodes that
+    had implemented fixes for fake funding, was able to trigger a [race
+    condition][] that crashed CLN with about 30 seconds of effort.  If
+    an LN node is offline, it can't defend a user against malicious or
+    broken counterparties, which puts the user's funds at risk.
+    Analysis indicated that CLN had fixed the original fake funding
+    vulnerability but was unable to safely include a test for it before
+    the vulnerability was disclosed, resulting the subsequent merge of a
+    plugin introducing the exploitable race condition.  After
+    Morehouse's disclosure, a quick patch was merged in CLN to prevent
+    the race condition from crashing the node.
+
+    For more information, we recommend reading Morehouse's excellent
+    [full disclosure][morehouse full] blog post.
+
+- **New LNHANCE combination soft fork proposed:** Brandon Black
+  [posted][black lnhance] to Delving Bitcoin details about a soft fork
+  that combines previous proposals for [OP_CHECKTEMPLATEVERIFY][topic
+  op_checktemplateverify] (CTV) and [OP_CHECKSIGFROMSTACK][topic
+  op_checksigfromstack] (CSFS) with a new proposal for an
+  `OP_INTERNALKEY` that places the taproot internal key on the stack.
+  Authors of scripts must know the internal key before they can pay to
+  an output, so they could directly include an internal key in a script.
+  However, `OP_INTERNALKEY` uses an old suggestion <!-- FIXME:cite -->
+  from CTV's original author, Jeremy Rubin, to save several vbytes and
+  potentially make scripts more reusable by allowing the value of the
+  key to be retrieved from the script interpreter.
+
+    In the thread, Black and others describe some of the protocols that
+    would be enabled by this combination of consensus changes:
+    [LN-Symmetry][topic eltoo] (eltoo), [Ark][topic ark]-style
+    [joinpools][topic joinpools], reduced-signatures [DLCs][topic dlc],
+    and [vaults][topic vaults] without presigned transactions, among
+    other described benefits of the underlying proposals, such as
+    CTV-style congestion control and CSFS-style signature delegation.
```

Comment:
Ugh, congestion control is such an absurd application, I really don’t understand why people keep bringing that up.

Reply:
It's still the first thing described in BIP119:

> This BIP proposes a new opcode, OP_CHECKTEMPLATEVERIFY, to be activated as a change to the semantics of OP_NOP4.
>
> The new opcode has applications for transaction congestion control and [...]

(But I also tend to think that it's impractical.)